src/mono/wasm/runtime/corebindings.c

```diff
@@ -165,11 +165,13 @@ mono_wasm_unbox_enum (MonoObject *obj)
 		return *(int*)ptr;
 	case MONO_TYPE_U4:
 		return *(unsigned int*)ptr;
-	// WASM doesn't support returning longs to JS
-	// case MONO_TYPE_I8:
-	// case MONO_TYPE_U8:
+	// WASM doesn't support returning longs to JS i.e. JavaScript does not have a native 64-bit int
+	case MONO_TYPE_I8:
```

Comment:
Can we add a test for this?

Reply:
The tests are part of https://github.com/dotnet/runtime/pull/41081

Reply:
Just wondering - https://github.com/kjpou1/runtime/blob/94323291a0bd078cd57fa4f72e1f525c836b1da7/src/mono/wasm/runtime/binding_support.js#L697-L697 - is this related? should it be removed/changed?

Reply:
Can be but prefer other PR.